python/lsst/obs/base/ingest.py

```diff
@@ -168,15 +168,16 @@ class RawExposureData:
     """Set of all known dimensions.
     """
 
-    record: Optional[DimensionRecord] = None
+    record: DimensionRecord
     """The exposure `DimensionRecord` that must be inserted into the
     `~lsst.daf.butler.Registry` prior to file-level ingest (`DimensionRecord`).
     """
 
-    def __post_init__(self, universe: DimensionUniverse) -> None:
-        # We don't care which file or dataset we read metadata from, because
-        # we're assuming they'll all be the same; just use the first ones.
-        self.record = makeExposureRecordFromObsInfo(self.files[0].datasets[0].obsInfo, universe)
+    extraRecords: Dict[str, DimensionRecord]
+    """Additional records that must be inserted into the
+    `~lsst.daf.butler.Registry` prior to ingesting the exposure ``record``
+    (e.g., to satisfy foreign key constraints), indexed by the dimension name.
+    """
```

Comment:
I suspect the structure of this requires that the extra records be one-to-one with `exposure`.  Would it be worthwhile to make this `Dict[str, Set[DimensionRecord]]` to generalize it to support e.g. `exposure`+`detector` records?